### ISSUE TITLE
Installation issue: nvhpc

### ISSUE THREAD

**Issue by lisanhu** [LLNL/spack#32022]
### Steps to reproduce the issue

```console
lisanhu@pop-os:~/mine/prog/spack$ spack find
==> 0 installed packages
lisanhu@pop-os:~/mine/prog/spack$ spack install nvhpc
==> Installing nvhpc-22.7-d2yk5h6qs4hpaesa5cqrlgp5xe3pkqsz
==> No binary for nvhpc-22.7-d2yk5h6qs4hpaesa5cqrlgp5xe3pkqsz found: installing from source
==> Using cached archive: /home/lisanhu/mine/prog/spack/var/spack/cache/_source-cache/archive/3c/3ce1c346f8bc7e50defb41c545c8907fdc012ff60b27eb8985cf3213f19d863a.tar.gz
==> No patches needed for nvhpc
==> nvhpc: Executing phase: 'install'
==> Error: TypeError: expected string or bytes-like object

/home/lisanhu/mine/prog/spack/var/spack/repos/builtin/packages/nvhpc/package.py:321, in install:
        318        install()
        319
        320        # Update localrc to use Spack gcc
  >>    321        makelocalrc(*makelocalrc_args)

See build log for details:
  /tmp/lisanhu/spack-stage/spack-stage-nvhpc-22.7-d2yk5h6qs4hpaesa5cqrlgp5xe3pkqsz/spack-build-out.txt

...
```

### Error message

<details><summary>Error message</summary><pre>
==> Error: TypeError: expected string or bytes-like object

/home/lisanhu/mine/prog/spack/var/spack/repos/builtin/packages/nvhpc/package.py:321, in install:
        318        install()
        319
        320        # Update localrc to use Spack gcc
  >>    321        makelocalrc(*makelocalrc_args)

See build log for details:
  /tmp/lisanhu/spack-stage/spack-stage-nvhpc-22.7-d2yk5h6qs4hpaesa5cqrlgp5xe3pkqsz/spack-build-out.txt

</pre></details>

### Information on your system

* **Spack:** 0.19.0.dev0 (8598da337da5fd580f7ad301e192a5e93006329e)
* **Python:** 3.10.4
* **Platform:** linux-pop22-zen3
* **Concretizer:** clingo

### Additional information

[spack-build-01-install-out.txt](https://github.com/spack/spack/files/9296545/spack-build-01-install-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/9296546/spack-build-env.txt)
[spack-build-env-mods.txt](https://github.com/spack/spack/files/9296547/spack-build-env-mods.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/9296548/spack-build-out.txt)
@samcmill

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by lisanhu** [comment]
something to add:
My system is a newly installed pop os 22.04 (ubuntu-based distribution).

After commenting line 321, it could install successfully. I've put a line `print(makelocalrc, makelocalrc_args)` in `spack edit nvhpc` to check what happened.

```
/home/lisanhu/mine/prog/spack/opt/spack/linux-pop22-zen3/gcc-11.2.0/nvhpc-22.7-qzymjm2vk3pqvpdicunzojimyt5naxby/Linux_x86_64/22.7/compilers/bin/makelocalrc ['-gcc', '/usr/bin/gcc', '-gpp', '/usr/bin/g++', '-g77', None, '-x', '/home/lisanhu/mine/prog/spack/opt/spack/linux-pop22-zen3/gcc-11.2.0/nvhpc-22.7-qzymjm2vk3pqvpdicunzojimyt5naxby/Linux_x86_64/22.7/compilers/bin']
```

It seems to call makelocalrc binary in the not yet installed path.

**Comment by samcmill** [comment]
This looks like the problem: `'-g77', None,`.

Do you have the GNU Fortran compiler installed?

**Comment by lisanhu** [comment]
I didn't have the fortran compiler.
I just set up a virtual machine and tested with and without gfortran, and the results are the same :( 
It's still showing the same error message.
I'm on a newly installed pop os 22.04 just FYI.

**Comment by Mr-Jeffery** [comment]
I also have the same problem. Did you find any solution to solve it?

**Comment by lisanhu** [comment]
I just tested it on a Ubuntu docker image, and the problem is still there. However, I managed to find a workaround this time. Let me tell you the workaround first then I'll discuss the causes.
spack version: 0.20.0.dev0 (ddea33bdc0a5cd14ecec700d552455a2ec7ca4de)
1. install gfortran or other fortran compiler in the system
2. run `spack compiler find` (mine says found no new compilers even though I just installed gfortran)
3. go to edit the compiler definition file (mine's at /root/.spack/linux/compilers.yaml), and change the line `f77: null` to `f77: {path_to_your_fortran_compiler}`, and remember to save it
4. run `spack install nvhpc` again

The problem is still about the makelocalrc executable, it receives -g77 None as the command line parameter which causes the problem.

The problem is not solved by installing the gfortran compiler, and spack compiler find is not really working for fortran compiler, even though I saw gfortran defined in Gcc class.

I find a similar issue for openmpi package at https://github.com/spack/spack/issues/7147
I agree that this is not an issue for openmpi package definition or nvhpc package difinition, but I think it's an issue for spack compiler find command.

Shall I close this issue and create another one for spack compiler find?
